diff -uprN automake-1.16.5/bin/automake.in automake-1.16.5-modified/bin/automake.in
--- automake-1.16.5/bin/automake.in	2021-09-20 09:53:14
+++ automake-1.16.5-modified/bin/automake.in	2024-02-09 16:53:08
@@ -739,6 +739,7 @@ register_language ('name' => 'objc',
 		   'compiler' => 'OBJCCOMPILE',
 		   'compile_flag' => '-c',
 		   'output_flag' => '-o',
+		   'libtool_tag' => 'CC',
 		   'lder' => 'OBJCLD',
 		   'ld' => '$(OBJC)',
 		   'pure' => 1,
@@ -757,6 +758,7 @@ register_language ('name' => 'objcxx',
 		   'compiler' => 'OBJCXXCOMPILE',
 		   'compile_flag' => '-c',
 		   'output_flag' => '-o',
+		   'libtool_tag' => 'CXX',
 		   'lder' => 'OBJCXXLD',
 		   'ld' => '$(OBJCXX)',
 		   'pure' => 1,
